.github/workflows/main.yml

```diff
@@ -50,7 +56,9 @@ jobs:
           script: |
             // Never skip workflow runs for pull requests, merge groups, or manually triggered
             // workfows / try jobs, which might need to actually run / retry WPT tests.
-            if (!['issue_comment', 'merge_group', 'pull_request', 'workflow_run', 'workflow_call'].includes(context.eventName)) {
+            let isPushToMaster = context.eventName == "push" && process.env.GITHUB_REF_NAME == "master";
+            let isTryRun = context.eventName == 'issue_comment' || (context.eventName == 'push' && !isPushToMaster);
+            if (!isTryRun && !['merge_group', 'pull_request', 'workflow_run', 'workflow_call'].includes(context.eventName)) {
               // Skip the run if an identical run already exists. This helps to avoid running
               // the workflow over and over again for the same commit hash.
               if ((await github.rest.actions.listWorkflowRuns({
```

Comment:
I'm having a hard time figuring out when this if arm will be taken. Which events are being considered for de-deduplication here?

Reply:
Sure I can add a comment here. How does something like this sound?

```
            // Skip this workflow if an identical run already exists. Note that we don't always want to skip workflows. In
            // particular we don't skip duplicate workflows if:
            //   1. This is a try job (triggered via an `issue_comment` event or a push to a non-master branch)
            //   2. This is a merge queue event (`merge_queue`)
            //   3. This event was triggered by a pull request update (`pull_request`)
            //   4. This event was triggered manually `workflow_run` / `workflow_call`
            let isPushToMaster = context.eventName == "push" && process.env.GITHUB_REF_NAME == "master";
            let isTryRun = context.eventName == 'issue_comment' || (context.eventName == 'push' && !isPushToMaster);
            if (!isTryRun && !['merge_group', 'pull_request', 'workflow_run', 'workflow_call'].includes(context.eventName)) {
```

Honestly, we can likely just remove this code which prevents duplicate workflow runs now. It was more important when bors was triggering try runs for commits that we were not testing.

Reply:
> Honestly, we can likely just remove this code which prevents duplicate workflow runs now. It was more important when bors was triggering try runs for commits that we were not testing.

That was my suspicion as well, but I wasn't able to follow the negated clauses to figure out which events do actually trigger the skip-workflow logic here. I'm fine with removing the logic if you think that is safe to do so.

For the comment, the new comment is similar to the existing one above the if statement.  I guess I was looking an example of an event where the skip-workflow logic is still executed.

Reply:
That's a good point. I believe that this mainly skips jobs for pushes to master. I agree this is a bit of a weird edge case and confusing. I'll will post a followup that just removes this for now.